presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java

```diff
@@ -302,6 +306,12 @@ public PlanNode getRight()
         return spillable;
     }
 
+    @JsonProperty("canSkipOutputDuplicates")
+    public boolean isCanSkipOutputDuplicates()
```

Comment:
This will affect the computed estimates for the join, but I don't see anything in this PR that adjusts that logic. The resulting cardinality should be no larger than that of the left side of the join.

Reply:
Currently, `OptimizeOutputDuplicatesInsensitiveJoinRule` is executed after `ReorderJoins` since `ReorderJoins` creates new `JoinNodes` without `canSkipOutputDuplicates` property.

I think, we could improve `ReorderJoins` to understand if entire join-subtree `canSkipOutputDuplicates`, but this is not part of this PR.